.appveyor.yml

```
@@ -238,9 +238,9 @@ install:
   # appveyor provided environments/installation). Note, these are broken
   # on the ubuntu images
   # https://help.appveyor.com/discussions/problems/28217-appveyor-ubunu-image-with-python3-lzma-module
-  # Otherwise create a virtualenv using the default Python 3, to enable uniform
+  # Otherwise create a new venv using the default Python 3, to enable uniform
   # use of python/pip executables below
-  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || virtualenv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
+  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python -m venv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```

Should it not be `python3`? `python` resolves to a Python interpreter with version 2.7.18 in an ssh session in `Ubuntu2004`.  

```suggestion
  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python3 -m venv -p 3 ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```

---

yes! thanks for spotting @christian-monch !  I misread the code and assumed that we would be after effects of `. ${HOME}/venv${PY}/bin/activate` when we would be having that python to be python3.  So indeed some old systems could have python2 still be installed  so let's ensure that we have python3, but then `-p 3` is not needed and I think is not supported by venv since we execute it via `python3` itself anyways hence

```suggestion
  - sh: "[ \"x$PY\" != x ] && . ${HOME}/venv${PY}/bin/activate || python3 -m venv ${HOME}/dlvenv && . ${HOME}/dlvenv/bin/activate; ln -s \"$VIRTUAL_ENV\" \"${HOME}/VENV\""
```